.github/workflows/main.yml

```diff
@@ -68,7 +68,34 @@ jobs:
           export LD_LIBRARY_PATH=$HOME/.wasmedge/lib
           cd ./crates && cargo test --all --all-features --no-fail-fast
       - name: Run feature tests
-        run: just featuretest
+        run: just test-features
+
+  # musl target testing is religated to a separate job because our current build
+  # for musl requires nightly. It is difficult to mix stable and nightly
+  # toolchains in the same job.
+  musl:
+    needs: [changes]
+    if: needs.changes.outputs.any_modified == 'true'
+    runs-on: ubuntu-22.04
+    timeout-minutes: 15
+    steps:
+      - uses: actions/checkout@v3
+      # We do not use `actions-rust-lang/setup-rust-toolchain` here because we
+      # want to override the default toolchain file to use nightly toolchains.
+      # The `setup-rust-toolchain` action will always choose toolchain file with
+      # no way to override. 
+      - name: Setup Rust toolchain and cache
+        uses: dtolnay/rust-toolchain@stable
```

Comment:
Do we need to specify `stable`? I wonder because we have had `rust-toolchain.toml`.

Reply:
Good catch! I mis-read the documentation and thought the `stable` referred to the github action version... I will make the change.